DRAFT Alfa java course (#1844)
CodeYourFuture/curriculum: common-content/en/module/java/sprint-1/section-1/index.md

+++
title = 'Section 1: Getting Started with the IDE and main methods'

time = 120
[objectives]
    1='Install and set up IntelliJ IDEA.'
    2='Navigate the IDE interface: project explorer, editor, console, and debugger.'
    3='Create a basic Java project and run a simple program.'

>     4='Explain the role of the main method in Java applications.'

Reviewer: Personally I tend to write these as:

```suggestion
objectives = [
    'Install and set up IntelliJ IDEA.',
    'Navigate the IDE interface: project explorer, editor, console, and debugger.',
    'Create a basic Java project and run a simple program.',
    'Explain the role of the main method in Java applications.',
]
```

to avoid needing to re-number things in the future

[build]
  render = 'never'
  list = 'local'
  publishResources = false

+++

> # Section 1: Getting Started with the IDE and main methods

Reviewer: Let's drop the duplicate titles

```suggestion
```

## Self Study
As you read through the resources below try to answer the following questions:
1.	You've probably used an IDE before when coding in other languages - what makes an IDE particularly useful when coding in Java? 
2.	What is a Java main method for? 
3.	What are some differences between how a program is run in Java vs other languages you've seen before?

### Reading material 
- [Introduction to Java](https://www.w3schools.com/java/java_intro.asp)
- [Interpreted vs Compiled Programming Languages](https://www.freecodecamp.org/news/compiled-versus-interpreted-languages/)
- [Getting started with IntelliJ IDEA](https://www.jetbrains.com/help/idea/getting-started.html)
- [Top 5 reasons you need an IDE](https://medium.com/@synergenttechsolution30/java-ides-top-5-reasons-you-need-an-ides-95426a0e6d5a)
- [Java main() Method Explained](https://www.baeldung.com/java-main-method)

### Exercise 1.1
Goal: Get comfortable with IntelliJ and running Java code.
- Install IntelliJ IDEA and create a new Java project.
- Create a main method and add logic so that it prints: "Hello, Java!"
- You'll need to Google for how to print to the console in Java!

> - Run the program from the IDE.

Reviewer: ```suggestion
{{<note type="exercise" title="Exercise 1.1">}}
Goal: Get comfortable with IntelliJ and running Java code.
- Install IntelliJ IDEA and create a new Java project.
- Create a main method and add logic so that it prints: "Hello, Java!"
- You'll need to Google for how to print to the console in Java!
- Run the program from the IDE.
- 
{{</note>}}
```
